mailserver/mailserver.go

```diff
@@ -380,41 +409,38 @@ func (s *WMailServer) openEnvelope(request *whisper.Envelope) *whisper.ReceivedM
 }
 
 // validateRequest runs different validations on the current request.
-func (s *WMailServer) validateRequest(peerID []byte, request *whisper.Envelope) (bool, uint32, uint32, []byte, uint32, cursorType) {
+func (s *WMailServer) validateRequest(peerID []byte, request *whisper.Envelope) (error, uint32, uint32, []byte, uint32, cursorType) {
 	if s.pow > 0.0 && request.PoW() < s.pow {
-		return false, 0, 0, nil, 0, nil
+		return fmt.Errorf("PoW() is too low"), 0, 0, nil, 0, nil
 	}
 
 	decrypted := s.openEnvelope(request)
 	if decrypted == nil {
-		log.Warn("Failed to decrypt p2p request")
-		return false, 0, 0, nil, 0, nil
+		return fmt.Errorf("Failed to decrypt p2p request"), 0, 0, nil, 0, nil
```

Comment:
The usual practice is to return errors not capitalized so that it's easily readable when errors are wrapped one into another.